.github/workflows/deploy.yaml

```diff
@@ -18,7 +19,14 @@ jobs:
     steps:
     - name: Checkout repository
       uses: actions/checkout@v2
-
+    - name: Configure NASA Earthdata Login
+      if: ${{ 'EARTHDATA_USER' in secrets }}
+      env:
+        EARTHDATA_USER: ${{ secrets.EARTHDATA_USER}}
+        EARTHDATA_PASS: ${{ secrets.EARTHDATA_PASS }}
+      run: |
+        echo "machine urs.earthdata.nasa.gov login $EARTHDATA_USER password $EARTHDATA_PASS" > ~/.netrc
+        chmod 0600 ~/.netrc
```

Comment:
I know we've gone back and forth a few times on whether or not to leave this in the template. It seems like we are needing it for multiple events and might make sense to include; should it have a few lines in the how-to guide for setting up the secrets in a repo using the template? Happy to open as a separate issue/PR, but wanted to note it for now.

Reply:
I am on board with that.

Note that I added the ```      if: ${{ 'EARTHDATA_USER' in secrets }}``` condition to only executes this once the secrets are set. Not sure if they work and will require further testing

Reply:
looks a little more complicated to do this conditional. https://stackoverflow.com/questions/70249519/how-to-check-if-a-secret-variable-is-empty-in-if-conditional-github-actions . Another option is `continue-on-error: true` for that step. Even for non-nasa hackweeks it's maybe nice to keep as it demonstrates how to safely use secrets in CI...

Reply:
generally it's hard to test any workflow that is `pull_request_target` because it has to be merged first! You can create pull requests against your fork's main branch... but I'm fine merging here then doing follow-on PRs for fixes

Reply:
if the secrets dont exist, it just creates a ~/.netrc that looks like `machine urs.earthdata.nasa.gov login  password  `